### ISSUE TITLE
1.57.0 (macOS): window behaves different now (fullscreen & priority issues)

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#126025]
We are seeing increased number of issues on macOS regarding VSCode window not behaving like before. This is likely a regression from https://github.com/electron/electron/pull/29235/files

This is a revert to Electron `12.0.7` and a revert of a commit I had pushed as workaround for this issue.

**Workaround:** none today, you can go back to `1.56.2` via [download](https://az764295.vo.msecnd.net/stable/054a9295330880ed74ceaedda236253b4f39a335/VSCode-darwin-universal.zip) and configure `update.mode` to `none` or `manual`.

//cc @deepak1556

**Comment by ryenus** [comment]
> **Workaround**: none today, you can go back to 1.56.2 via download

Maybe it's obvious, I had to disable automatic update by setting `update.mode` to `none` or `manual`.

**Comment by jamesb93** [comment]
I use Alfred to move between applications and currently VSCODE will retake control of priority whenever I use their spotlight replacement to change applications.

**Comment by jacobweber** [comment]
I'm guessing this is related to why command-H no longer hides VS Code some of the time?

**Comment by jamesb93** [comment]
> I'm guessing this is related to why command-H no longer hides VS Code some of the time? Seems to behave differently depending on which workspace window is open. This is on 1.57.0.

 I always work in a workspace. Does it depend on whether or not you are in a workspace at all, or just on the workspace itself?

**Comment by jacobweber** [comment]
> I always work in a workspace. Does it depend on whether or not you are in a workspace at all, or just on the workspace itself?

I didn't think there a way to NOT be in a workspace, other than having all windows closed. But I'm generally working in single-folder workspaces. Anyway, it seems to depend on the particular window -- I have two open, and it's only happening when a certain one is in front. But if I close and reopen that one, it doesn't start happening for a while. If this upcoming recovery doesn't fix it, I'll try to narrow down the cause a little more.

**Comment by dennis-damico** [comment]
>VSCODE will retake control of priority

I'm having a similar issue trying to switch applications with [alt tab](https://alt-tab-macos.netlify.app/)

**Comment by tricktron** [comment]
I downgraded to `1.56.2` and started vs-code from command line with `code --disable-extensions .`

Result: I can't render/show any files anymore:
![error-screenshot](https://user-images.githubusercontent.com/16036882/122066667-3d0de580-cdf3-11eb-80f3-bb118671b7b4.png)

The following warnings are shown in the developer console:
```log
ERR (this.configurationService.getValue(...) || []).filter is not a function: TypeError: (this.configurationService.getValue(...) || []).filter is not a function
```

When I create a new file `test2.txt` I get the same error again.

Here you have the full [log](https://github.com/microsoft/vscode/files/6656195/log2.log).

I am on Big Sur 11.4 and

```log
Version: 1.56.2
Commit: 054a9295330880ed74ceaedda236253b4f39a335
Date: 2021-05-12T17:44:30.902Z (1 mo ago)
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Darwin x64 20.5.0
```

Does anyone else experience this issue?

**Comment by realappie** [comment]
Yeah this issue is frustrating, the native tabs feature seems to be broken in the last upgrade. Every time I open a new window I get a new macOS space automatically opened

![image](https://user-images.githubusercontent.com/10495548/122178707-1dc09800-ce87-11eb-8ac3-4786c2753c91.png)

**Comment by lehungdev** [comment]
> Yeah this issue is frustrating, the native tabs feature seems to be broken in the last upgrade. Every time I open a new window I get a new macOS space automatically opened
> 
> **Edit: Disabling the native full screen setting seems to fix it for me.**
> 
> ![image](https://user-images.githubusercontent.com/10495548/122178707-1dc09800-ce87-11eb-8ac3-4786c2753c91.png)

Yes, I had to revert to the old version because of this error.

**Comment by uphlewis** [comment]
I get the same issue using contexts to cmd + tab between windows. VSCode just switches back to the front again. It seems inconsistent, i.e sometimes it doesnt do it. Super strange.

```
Version: 1.57.0
Commit: b4c1bd0a9b03c749ea011b06c6d2676c8091a70c
Date: 2021-06-09T17:22:31.215Z
Electron: 12.0.9
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Darwin x64 19.6.0
```

**Comment by Borongaj** [comment]
On macOS : VSCode 1.57.1 start, black screen, and close...

**Comment by uphlewis** [comment]
Appears fixed on 1.57.1 thanks everyone
